Update instructions to install SenseHat in HassOs

denics opened the pull request:
Added instruction for using SenseHat with HassOs

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issues already close by bot:
  - #32692  
  - #5093 
  - #10511 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

denics committed:
Update sensehat.markdown

Added instruction for using SenseHat with HassOs

denics committed:
Update sensehat.markdown

solved some issues with standard text

denics committed:
Update sensehat.markdown

added language flag

frenck commented on source/_integrations/sensehat.markdown:
```diff
@@ -87,13 +87,13 @@ sensor:
     sensors:
       sensehat_temperature:
-        value_template: '{{ states('sensor.temperature') | round(1) }}'
+        value_template: '{{ states("sensor.temperature") | round(1) }}'
```
This is not according to our documentation standards. See: <https://developers.home-assistant.io/docs/documenting/standards#templates>

```suggestion
        value_template: "{{ states('sensor.temperature') | round(1) }}"
```

Please adjust the other cases below as well.

denics replied:
But this means the final user cannot do cut&paste of our example, which is confusing.

frenck replied:
Yes they can.

frenck commented on source/_integrations/sensehat.markdown:
````diff
@@ -140,8 +140,35 @@ group:
 ```
-
-### Directions for installing on Raspberry Pi Raspbian Based installation:
+## Installation
````
- Headers should be surrounded by empty new lines.
- "Installation" sounds confusing right here?

denics replied:
Done, instead of _Installation_ I used _Install on_

frenck commented on source/_integrations/sensehat.markdown:
```diff
@@ -140,8 +140,35 @@ group:
+## Installation
+### On Home Assistant Operating System
+
+Things got complicated for poor _SenseHAT_ with Home Assistant Operating System. I found finally a way to use it.
```
Your personal opinion is not something we should document.

```suggestion
Things got complicated for poor _SenseHAT_ with Home Assistant Operating System. I found finally a way to use it.
```

denics replied:
removed, I thought that a bit of humor could help

frenck commented on source/_integrations/sensehat.markdown:
```diff
@@ -140,8 +140,35 @@ group:
+- Install "Custom deps deployment" add-on. To do so
+    - add ``https://github.com/home-assistant/hassio-addons-development`` to your add-ons repositories
+    - select *Custom deps deployment* and enable it
+- Configure "Custom deps deployment" add-on
```
Custom deps is not recommend, we should not document that and/or writing instructions to use that.

denics replied:
we can add a warning if you think it is better, but this is the only way to make the integration works. Otherwise, we can remove the integration as it cannot be installed.

frenck replied:
A warning makes no sense.

If it doesn't work without these workarounds, I suggest filing an issue and get the problem fixed.

denics replied:
I am filing an issue in "home-assistant/operating-system" to include support for RTIMUlib. Can we leave it as "workaround" ? or we will have change this text to something like:

You will not be able to use your SenseHat until [issue 953](home-assistant/operating-system/issues/953) will not be solved.

What do you think?

frenck replied:
we don't document issues.

AgalmicVentures replied:
@frenck 

> Custom deps is not recommend, we should not document that and/or writing instructions to use that.
> we don't document issues.

The larger issue with the documentation is listing SenseHat as a supported device when it's not.

No need to be curt with @denics for trying to address the problem. Three tickets have already been filed and closed as stale. He's proposing two reasonable solutions to make this a better experience for your users.

If you honestly don't want to make it work, that's fine -- in that case, please remove SenseHat from your documentation so others don't waste time like I (and @denics) did.

frenck replied:
I'm sorry to hear about your issues. However, this is how it is. Feel free to contribute a fix @AgalmicVentures.

I'm fine with removing integrations if that is a better solution, especially if they are not maintained.

denics replied:
Dear @frenck , the issue here is not with the SenseHat integration, I even made a PR to make the integration more universal and I am ready to add a workflow to it. The problem is that HassOS does not provide a required library. In my "solution" I am suggesting a workaround if HassOS is not integrating the required library. I opened a ticket on HassOS to point that out and you moved it to core (I do not understand why).

frenck replied:
Because integration defines the installation requirements, not the OS.

If it can be installed by the custom deps add-on, it can be handled by defining dependencies in the integration. Both do the same: Installing Python packages.

Thus this is an integration issue (core), not the OS.

QuantumStar replied:
Gee, I sure wish I'd found this thread six hours ago. I wasted a ridiculous amount of time, thinking "Surely if it's in the documentation, the SenseHAT must work?". I appreciate that this is all done by volunteers, but like others have said, it would be ideal if we could just figure out how to get it working again, but if not REMOVE THE DAMN THING FROM THE DOCUMENTATION or list it as deprecated, something to save people like me from wasting their time.

frenck commented on source/_integrations/sensehat.markdown:
```diff
@@ -140,8 +140,35 @@ group:
+
+Things got complicated for poor _SenseHAT_ with Home Assistant Operating System. I found finally a way to use it.
+
+- Follow instructions [here](https://www.home-assistant.io/hassio/enable_i2c/) to enable i2c on Home Assistant Operating System
```
Don't use full URLs

denics replied:
what should I use then?

frenck replied:
From the root: `/hassio/enable_i2c`

frenck commented on source/_integrations/sensehat.markdown:
````diff
@@ -140,8 +140,35 @@ group:
+    - add ``https://github.com/home-assistant/hassio-addons-development`` to your add-ons repositories
+    - select *Custom deps deployment* and enable it
+- Configure "Custom deps deployment" add-on
+    ```yaml
````
Code fenced blocks need to be surrounded by an empty new line

denics replied:
solved.

frenck commented on source/_integrations/sensehat.markdown:
```diff
@@ -140,8 +140,35 @@ group:
+- Start the plugin and check the logs to make sure everything is fine. Set it to "Start at boot"
+- Reboot and enjoy your lights
+
+Steps verified with Home Assistant Operating System 4.15. At the moment of this writing, sensors of the SenseHat are still not working, but at least now they can be seen.
```
We don't document versions. Documentation should match the latest codebase.

denics replied:
Solved

denics replied:
However I find confusing and having a negative impact the fact of not documenting version, if you do not write the date of the documentation. In this specific case, the documentation is 3 years old (and the previous text mention a version) and we have at least 3 git issues on this item without solution (until now) and few others on the forum on the same subject. Maybe you can show in the right column not only the version since a module is maintained, but the last date of modification of the page. That would really help.

frenck replied:
We don't do that. Our documentation reflects the latest version and that is it.
At this point in time, keeping up with the current state of affairs is already hard enough. Maybe if we master that.

However, out of scope for this PR.

denics replied:
Thanks

denics committed:
Update sensehat.markdown

resolve two comments:



- Headers should be surrounded by empty new lines.
- "Installation" sounds confusing right here?

denics committed:
Update sensehat.markdown

corrected a comment

denics committed:
Update sensehat.markdown

resolve comment

denics committed:
Update sensehat.markdown

fixed comment

denics committed:
Update sensehat.markdown

resolved comment

denics committed:
Update sensehat.markdown

sovled comment

btwebz commented:
Hi all,
I'd really like to get my sensehat working too!
Would be great if someone could post some instructions on how to achieve this!
Thanks heaps

frenck commented:
@btwebz That is what this PR and discussion is about.

svanlee commented:
New user: need instructions on how to install on hass.io please

QuantumStar commented:
Guess nobody ever bothered to fix this, or update the documentation. What a huge waste of time...
